ksql-functional-tests/src/main/java/io/confluent/ksql/test/tools/TestCase.java

```diff
@@ -204,15 +210,29 @@ public void verifyOutput(final TopologyTestDriverContainer topologyTestDriverCon
             expectedOutput.value,
             expectedOutput.timestamp);
       }
-    } catch (final AssertionError assertionError) {
+    } catch (final AssertionError e) {
       final String rowMsg = idx == -1 ? "" : " while processing output row " + idx;
       final String topicMsg = idx == -1 ? "" : " topic: " + outputRecords.get(idx).topic.name;
-      throw new AssertionError("TestCase name: "
-          + name
-          + " in file: " + testPath
-          + " failed" + rowMsg + topicMsg + " due to: "
-          + assertionError.getMessage(), assertionError);
+      throw new AssertionError("failed" + rowMsg + topicMsg + " due to: "
+          + e.getMessage(), e);
     }
+
+    throwIfMoreOutputAvailable(topologyTestDriverContainer.getTopologyTestDriver());
+  }
+
+  private void throwIfMoreOutputAvailable(
+      final TopologyTestDriver driver
+  ) {
+    topics.forEach(topic -> {
+      final ProducerRecord<Bytes, Bytes> record = driver.readOutput(
+              topic.getName(),
+              new BytesDeserializer(),
+              new BytesDeserializer());
+
+      if (record != null) {
+        throw new AssertionError("Unexpected input available on topic: " + topic.name);
```

Comment:
nit: Unexpected data (input is a bit misleading in this context since there are both input and output topics)

Reply:
Good catch.